presto-kinesis/src/main/java/io/prestosql/plugin/kinesis/s3config/S3TableConfigClient.java

```diff
@@ -0,0 +1,266 @@
+                    "which means the client encountered " +
+                    "an internal error while trying to communicate" +
+                    " with S3, " +
+                    "such as not being able to access the network.");
+            sb.append("Error Message: " + ace.getMessage());
+            log.error(sb.toString(), ace);
+        }
+
+        return returnList;
+    }
+
+    /**
+     * Connect to S3 directory to look for new or updated table definitions and then
+     * update the map.
+     */
+    protected void updateTablesFromS3()
+    {
+        long now = System.currentTimeMillis();
+
+        List<S3ObjectSummary> objectList = this.getObjectSummaries();
+        AmazonS3Client s3client = this.clientManager.getS3Client();
+        AmazonS3URI directoryURI = new AmazonS3URI(this.bucketUrl);
+
+        for (S3ObjectSummary objInfo : objectList) {
+            if (!this.internalMap.containsKey(objInfo.getKey()) || objInfo.getLastModified().getTime() >= this.lastCheck) {
+                // New or updated file, so we must read from AWS
+                try {
+                    if (objInfo.getKey().endsWith("/")) {
+                        continue;
+                    }
+
+                    log.info("Getting : %s - %s", objInfo.getBucketName(), objInfo.getKey());
+                    S3Object object = s3client.getObject(
+                            new GetObjectRequest(objInfo.getBucketName(), objInfo.getKey()));
+
+                    StringBuilder resultStr = new StringBuilder("");
+                    try (BufferedReader reader = new BufferedReader(new InputStreamReader(object.getObjectContent()))) {
+                        boolean hasMore = true;
+                        while (hasMore) {
+                            String line = reader.readLine();
+                            if (line != null) {
+                                resultStr.append(line);
+                            }
+                            else {
+                                hasMore = false;
+                            }
+                        }
+
+                        KinesisStreamDescription table = streamDescriptionCodec.fromJson(resultStr.toString());
+
+                        internalMap.put(objInfo.getKey(), table);
+                        log.info("Put table description into the map from %s", objInfo.getKey());
+                    }
+                    catch (IOException iox) {
+                        log.error("Problem reading input stream from object.", iox);
+                    }
+                }
+                catch (AmazonServiceException ase) {
+                    StringBuilder sb = new StringBuilder();
+                    sb.append("Caught an AmazonServiceException, which means your request made it ");
+                    sb.append("to Amazon S3, but was rejected with an error response for some reason.\n");
+                    sb.append("Error Message:    " + ase.getMessage());
```

Comment:
Based on the aligned formatting, these all need newlines in the output string. However, we don't usually use this style of logging in Presto. It would be more typical to do it like this:
```java
log.error(ase, "Request to S3 was rejected");
```
The default `toString()` for `AmazonServiceException` already includes all this information (in a nice one-line format), so we don't need to reformat it ourselves.

Reply:
changed to this